libs/domain/search/facet/controllers/facet.controller.ts

```diff
@@ -1,49 +1,71 @@
 import { resolve } from '@spryker-oryx/di';
-import { Facet, FacetValue } from '@spryker-oryx/product';
+import { FacetValue } from '@spryker-oryx/product';
 import { FacetListService } from '@spryker-oryx/search';
 import {
   FACET_CLEAR_EVENT,
   FACET_TOGGLE_EVENT,
   ToggleFacetPayload,
 } from '@spryker-oryx/search/facet-value-navigation';
-import { ObserveController } from '@spryker-oryx/utilities';
+import { SearchPayload } from '@spryker-oryx/ui/searchbox';
+import { computed, ObserveController, signal } from '@spryker-oryx/utilities';
 import { LitElement, ReactiveController } from 'lit';
+import { Observable, of, switchMap } from 'rxjs';
 import {
-  BehaviorSubject,
-  combineLatest,
-  iif,
-  map,
-  mergeMap,
-  Observable,
-  of,
-  switchMap,
-} from 'rxjs';
-import {
-  FacetComponentAttributes,
-  FacetSelect,
   FACET_SELECT_EVENT,
+  SearchFacetComponentAttributes,
+  SelectFacetPayload,
   SingleMultiFacet,
 } from '../facet.model';
 
 export class FacetController implements ReactiveController {
-  protected observe: ObserveController<LitElement & FacetComponentAttributes>;
+  protected observe = new ObserveController(this.host);
   protected facetListService = resolve(FacetListService);
-  protected facet$: Observable<Facet | null>;
-  protected showAll$ = new BehaviorSubject(false);
-  protected searchedValue$ = new BehaviorSubject('');
 
-  constructor(protected host: LitElement & FacetComponentAttributes) {
+  protected facet$ = this.observe.get('name').pipe(
```

Comment:
we should not need to observe now that name is a signalProperty

Reply:
But this is controller, in theory it can apply to any other facet-like component. Lets continue observing to keep it a bit more generic and flexible